src/ConnectedMode.UnitTests/ServerSentEvents/SSESessionTests.cs

```diff
@@ -193,8 +193,25 @@ public async Task Dispose_FinishesSession()
         sessionToken.Value.IsCancellationRequested.Should().BeTrue();
     }
 
+    [TestMethod]
+    public void Dispose_InvokesOnSessionFailedAsync()
```

Comment:
This seems weird - the session should be disposed e.g. when a bound solution is closed. Do we want to log a failure message and try to create another session in that case?

Reply:
I asked that in a comment above :)